overlord/snapstate/backend/snapdata.go

```diff
@@ -131,9 +137,14 @@ func snapDataDirs(snap *snap.Info, opts *dirs.SnapDirOptions) ([]string, error)
 // snapCommonDataDirs returns the list of data directories common between versions of the given snap
 func snapCommonDataDirs(snap *snap.Info, opts *dirs.SnapDirOptions) ([]string, error) {
 	// collect the directories, homes first
-	found, err := filepath.Glob(snap.CommonDataHomeDir(opts))
-	if err != nil {
-		return nil, err
+	var found []string
+
+	for _, entry := range snap.CommonDataHomeDirs(opts) {
+		entryPaths, err := filepath.Glob(entry)
```

Comment:
this part doesn't seem to be covered, could you please add/update unit tests to exercise this part? check `TestRemoveSnapCommonData` for reference.